modules/publication/ajax/FileDelete.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Publication file deletion script
+ *
+ * Deletes a specified file by upload ID (PublicationUploadID)
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Publication
+ * @author   David <dblader.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+$uploadID = $_REQUEST['uploadID'];
+$db       = \Database::singleton();
+$user     = \User::singleton();
+$config   = \NDB_Config::singleton();
+
+$query      = "SELECT PublicationID, URL ".
+    "FROM publication_upload ".
+    "WHERE PublicationUploadID=:upid";
+$uploadData = $db->pselectRow($query, array('upid' => $uploadID));
+
+if (empty($uploadData)) {
+    header("HTTP/1.1 400 Bad Request");
+    return;
+}
+
+$origUser = $db->pselectOne(
+    'SELECT UserID FROM publication WHERE PublicationID=:pid',
+    array('pid' => $uploadData['PublicationID'])
+);
+
+$editors = $db->pselectCol(
+    'SELECT UserID FROM publication_users_edit_perm_rel WHERE PublicationID=:pid',
+    array('pid' => $uploadData['PublicationID'])
+);
+
+if ($user->getId() !== $origUser && !in_array($user->getId(), $editors)) {
+    header("HTTP/1.1 403 Forbidden");
+    return;
+}
+
+$db->delete(
+    'publication_upload',
+    array('PublicationUploadID' => $uploadID)
+);
+
+$base = $config->getSetting('publication_uploads');
+
+unlink($base . $uploadData['URL']);
```

Comment:
Lines 40-43 ensures that the only person that can delete a file is either the original uploader OR one of the publication editors.

An additional module-level permissions check could be beneficial just in case but this feels OK to me.

Reply:
the thing that scares me about the line is that it unrecoverably deletes the file (and anyone listed as an editor can do it).

I agree that it's not a security problem because of those lines, but all scripts in an `ajax/` directory need to start with an explicit permission check (Jordan had went in and manually did it for all existing scripts a while ago, if there are any that are currently missing it's a regression..), the AjaxHelper.php does minimal checking itself and then just require's the file from the ajax directory. It should be treated as if it was just a random php script in the htdocs directory.